std/internal/math/gammafunction.d

```diff
@@ -531,8 +543,17 @@ unittest {
 
 
 private {
-enum real MAXLOG = 0x1.62e42fefa39ef358p+13L;  // log(real.max)
-enum real MINLOG = -0x1.6436716d5406e6d8p+13L; // log(real.min*real.epsilon) = log(smallest denormal)
+static if (floatTraits!(real).realFormat == RealFormat.ieeeExtended) {
+    enum real MAXLOG = 0x1.62e42fefa39ef358p+13L;  // log(real.max)
+    enum real MINLOG = -0x1.6436716d5406e6d8p+13L; // log(real.min_normal*real.epsilon) = log(smallest denormal)
+}
+else static if (floatTraits!(real).realFormat == RealFormat.ieeeDouble) {
+    enum real MAXLOG = 0x1.62e430p+9L;  // log(real.max)
+    enum real MINLOG = -0x1.743854p+9L; // log(real.min_normal*real.epsilon) = log(smallest denormal)
```

Comment:
Also you can find implementation of gamma for `double` and `float` in the CEPHES library. Current implementation was ported from CEPHES too.